src/lib/graphqlTimeoutMiddleware.ts

```diff
@@ -65,9 +65,10 @@ export const graphqlTimeoutMiddleware = (defaultTimeoutInMS: number) => {
           )
         }, timeoutInMS)
       }),
-      new Promise((resolve, reject) => {
+      // eslint-disable-next-line no-async-promise-executor
+      new Promise(async (resolve, reject) => {
         try {
-          const result = middlewareResolver(parent, args, context, info)
+          const result = await middlewareResolver(parent, args, context, info)
```

Comment:
Kinda curious about this one... how did it ever work?

Reply:
I guess you can use an async function as a promise executor. [It's just not encouraged because of error handling, and it shouldn't be necessary because the async function already returns a promise. ](https://eslint.org/docs/latest/rules/no-async-promise-executor)

Probably, the idea here was to be able to use the[ finally block](https://github.com/artsy/metaphysics/blob/f48040d9d8a0c42e6e6aeafca948ab08ec0295d8/src/lib/graphqlTimeoutMiddleware.ts#L75) to clear the timeout 🤔

Reply:
Ahh right! That makes sense 👍